pkg/networkservice/common/timeout/server_test.go

```diff
@@ -121,13 +132,17 @@ func TestTimeoutServer_CloseBeforeTimeout(t *testing.T) {
 	require.Condition(t, connServer.validator(0, 1))
 
 	// ensure there will be no double Close
-	<-time.After(waitFor)
+	clockMock.Add(tokenTimeout)
+	time.Sleep(testWait)
```

Comment:
Why do we need to use real time.Sleep together with a mock?

Reply:
There are 2 types of waitings in the tests we need:
1. waiting for the timeout to happen (timeout, refresh, expire, retry after, ... cases)
2. waiting for the event to pass the chain

`clock` makes unnecessary 1 kind of waitings, but not 2. So we still need for `require.Eventually` or `time.Sleep` to be sure that event happens or not happens.